cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -492,6 +486,81 @@ TYPED_TEST(TestBinaryArithmeticFloating, Add) {
+
+    this->AssertBinop(Divide, "[1.1, 2.4, 3.5, 4.3, 5.1, 6.8, 7.3]",
+                      "[1.0, 2.0, 0.7, 0.5, 1.7, 2.0, 5.0]",
+                      "[1.1, 1.2, 5.0, 8.6, 3.0, 3.4, 1.46]");
+
+    this->AssertBinop(Divide, "[10.4, 12, 4.2, 50, 50.5, 32, 11]",
+                      "[2.0, 1.0, 6, 1, 5, 8, 2]", "[5.2, 12, 0.7, 50, 10.1, 4, 5.5]");
+
+    this->AssertBinop(Divide, "[null, 1, 3.3, null, 2, 5.1]", "[1, 4, 2, 5, 0.1, 3]",
+                      "[null, 0.25, 1.65, null, 20, 1.7]");
+
+    this->AssertBinop(Divide, 10.0F, "[null, 1, 2.5, null, 2, 5]",
+                      "[null, 10, 4, null, 5, 2]");
+
+    this->AssertBinop(Divide, "[null, 1, 2.5, null, 2, 5]", 10.0F,
+                      "[null, 0.1, 0.25, null, 0.2, 0.5]");
+
+    this->AssertBinop(Divide, 21.0F, 3.0F, 7.0F);
+  }
+}
+
+TYPED_TEST(TestBinaryArithmeticIntegral, Div) {
+  for (auto check_overflow : {false, true}) {
+    this->SetOverflowCheck(check_overflow);
+
+    this->AssertBinop(Divide, "[]", "[]", "[]");
+    this->AssertBinop(Divide, "[null]", "[null]", "[null]");
+    this->AssertBinop(Divide, "[3, 2, 6]", "[1, 1, 2]", "[3, 2, 3]");
+    this->AssertBinop(Divide, "[10, 20, 30, 40, 50, 60, 70]", "[10, 5, 2, 1, 25, 30, 35]",
+                      "[1, 4, 15, 40, 2, 2, 2]");
+    this->AssertBinop(Divide, "[70, 60, 50, 40, 30, 20, 10]", "[7, 15, 5, 2, 6, 4, 5]",
+                      "[10, 4, 10, 20, 5, 5, 2]");
+    this->AssertBinop(Divide, "[null, 10, 30, null, 20, 50]", "[1, 4, 2, 5, 10, 3]",
+                      "[null, 2, 15, null, 2, 16]");
+    this->AssertBinop(Divide, 33, "[null, 1, 3, null, 2, 5]",
+                      "[null, 33, 11, null, 16, 6]");
+    this->AssertBinop(Divide, 16, 7, 2);
+  }
+}
+
+TYPED_TEST(TestBinaryArithmeticSigned, Div) {
+  this->AssertBinop(Divide, "[-3, 2, -6]", "[1, 1, 2]", "[-3, 2, -3]");
+  this->AssertBinop(Divide, "[10, 20, -30, 40, -50, 60, 70]", "[10, 5, 2, 1, 25, 30, 35]",
+                    "[1, 4, -15, 40, -2, 2, 2]");
+  this->AssertBinop(Divide, "[70, 60, 50, 40, 30, 20, 10]", "[7, -15, 5, 2, 6, 4, -5]",
+                    "[10, -4, 10, 20, 5, 5, -2]");
+  this->AssertBinop(Divide, "[null, 10, 30, null, -20, 50]", "[1, 4, 2, 5, 10, 3]",
+                    "[null, 2, 15, null, -2, 16]");
+  this->AssertBinop(Divide, 33, "[null, -1, -3, null, 2, 5]",
+                    "[null, -33, -11, null, 16, 6]");
+  this->AssertBinop(Divide, -16, -8, 2);
+}
+
+TYPED_TEST(TestBinaryArithmeticIntegral, DivideByZero) {
+  for (auto check_overflow : {false, true}) {
+    this->SetOverflowCheck(check_overflow);
+    this->AssertBinopRaises(Divide, "[3, 2, 6]", "[1, 1, 0]", "overflow");
+  }
+}
+
+TYPED_TEST(TestBinaryArithmeticSigned, DivideOverflowRaises) {
+  using CType = typename TestFixture::CType;
+
+  auto min = std::numeric_limits<CType>::lowest();
+  this->SetOverflowCheck(true);
+
+  this->AssertBinopRaises(Divide, MakeArray(min), MakeArray(-1), "overflow");
```

Comment:
Should also check what happens when the overflow checks are disabled?

Reply:
When overflow checks are disabled, we get SIGFP, and the tests crash.

Reply:
Hmm, we should definitly not crash on overflow. In this case, I think overflow should be detected and a dummy value be output (0 perhaps?).

Reply:
Sounds reasonable. I have updated the PR accordingly.